### PR TITLE
Display ENS name for connected wallet

### PR DESCRIPTION
**Pull request by rootulp**
## Motivation
ENS airdrop last week 🎉🥳 Let's make indexcoop.com ENS aware

## Description
1. Change "Unlock wallet" to "Connect wallet" because the latter seems more common on Web3 sites
2. Display ENS name for connected wallet (if user has an ENS). If not, fallback to shortened version of their public address.

Scenario | Screenshot
--- | ---
Not connected | ens-not-connected
Connected without ENS | ens-without-ens
Connected with ENS | ens-with-ens

## Bundle size

This introduces a new dependency on [useDapp](https://usedapp.readthedocs.io/en/latest/) which increases our bundle size by approx. (1.97MB - 1.89MB) = .08MB
Before | After
--- | ---
bundle-size-before | bundle-size-after

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -149,65 +154,67 @@ const Providers: React.FC = ({ children }) => {
...
-                          </MviStakingRewardsProvider>
-                        </FarmingTwoProvider>
-                      </FarmingProvider>
-                    </BalancesProvider>
-                  </ExternalAirdropProvider>
-                </RewardsProvider>
-              </AirdropProvider>
-            </MediaQueryProvider>
-          </ApolloProvider>
+          <DAppProvider config={config}>
```
Comment (rootulp) [inline src/App.tsx:157]: The only line change here is adding `<DAppProvider config={config}>`. Every line below just gets indented by one tab

File: src/components/TopBar/components/WalletButton.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import React, { useCallback } from 'react'
 import styled from 'styled-components'
-
 import { Button } from 'react-neu'
-import useWallet from 'hooks/useWallet'
+import { shortenAddress, useLookupAddress } from '@usedapp/core'
 
+import useWallet from 'hooks/useWallet'
```
Comment (rootulp) [inline src/components/TopBar/components/WalletButton.tsx:6]: grouped external / internal imports. Separated by newline

File: package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "@testing-library/jest-dom": "^5.15.0",
     "@testing-library/react": "^12.1.2",
     "@testing-library/user-event": "^13.5.0",
+    "@usedapp/core": "^0.5.5",
```
Comment (0xModene) [inline package.json:13]: Yes yes, the march toward ethers begins!
muahaha

File: src/App.tsx (hunk 1 above)
Comment (0xModene) [reply src/App.tsx:157]: soon we will kill some of these providers

**Review by 0xModene — APPROVED** [review]
lgtm!
I'll wait for #422 to merge first to be sure of no conflict, since that's a huge change

**Comment by 0xModene** [comment]
Actually going to go ahead and merge this in since I have to wait on @controtie to review #422

**Pull request merged**